core/src/main/scala/com/yahoo/maha/utils/GetTotalRowsRequest.scala

```diff
@@ -0,0 +1,64 @@
+package com.yahoo.maha.utils
+
+import com.yahoo.maha.core._
+import com.yahoo.maha.core.query._
+import com.yahoo.maha.core.query.druid.DruidQueryGenerator
+import com.yahoo.maha.core.query.oracle.OracleQueryGenerator
+import com.yahoo.maha.core.registry.Registry
+import com.yahoo.maha.core.request.{Field, ReportingRequest}
+import grizzled.slf4j.Logging
+
+import scala.util.Try
+
+object GetTotalRowsRequest extends Logging {
+
+  def getTotalRowsRequest(request: ReportingRequest, pipeline: QueryPipeline) : Try[ReportingRequest] = {
+    //no filters except fk filters
+    Try {
+      require(
+        pipeline.bestDimCandidates.nonEmpty
+        , s"Invalid total rows request, no best dim candidates! : $request")
+
+      //force dim driven
+      //remove all fields except primary key
+      //remove all sorts
+      val primaryKeyAliasFields = pipeline.bestDimCandidates.map(dim => Field(dim.publicDim.primaryKeyByAlias, None, None)).toIndexedSeq
+      request.copy(
+        selectFields = primaryKeyAliasFields
+        , sortBy = IndexedSeq.empty
+        , includeRowCount = true
+        , forceDimensionDriven = true
+        , forceFactDriven = false
+        , paginationStartIndex = 0
+        , rowsPerPage = request.rowsPerPage
+      )
+    }
+  }
+
+  def getTotalRows(request: RequestModel, sourcePipeline: QueryPipeline, registry: Registry, queryContext: QueryExecutorContext)(implicit queryGeneratorRegistry: QueryGeneratorRegistry) : Try[Int] = {
+    Try {
+      val totalRowsRequest: ReportingRequest = getTotalRowsRequest(request.reportingRequest, sourcePipeline).get
+      val model: RequestModel = RequestModel.from(totalRowsRequest, registry).get
+      val maxRows: Int = DruidQueryGenerator.defaultMaximumMaxRows
```

Comment:
Why are we using the value from DruidQueryGenerator here?  Do we even need this?  Aren't we expecting to just get 1 row returned with the # of total rows?